src/IRrecv.cpp

```diff
@@ -356,23 +371,35 @@ void IRrecv::enableIRIn(const bool pullup) {
     pinMode(params.recvpin, INPUT);
 #endif  // UNIT_TEST
   }
+
 #if defined(ESP32)
   // Initialise the ESP32 timer.
-  // 80MHz / 80 = 1 uSec granularity.
-  timer = timerBegin(_timer_num, 80, true);
+#if ARDUINO_COREV3
+  // Use newer timerBegin signature for ESP32 core version 3.x
+  timer = timerBegin(1000000);  // Initialize with 1MHz (1us per tick)
+#else
+  // Fallback for ESP32 core version 2.x or earlier
+  timer = timerBegin(0, 1000000, true);  // Old signature with divider
```

Comment:
Why the change from `timer = timerBegin(_timer_num, 80, true);` here?
could we leave it unchanged?

Reply:
with version 3 timerBegin support only one value instead of 3

Reply:
But this line is for V2 which is changed from `timer = timerBegin(_timer_num, 80, true);` to `timer = timerBegin(0, 1000000, true); `